### PR TITLE
Simple system info report for debug / support

### PR DESCRIPTION
**Pull request by PrzemekWirkus**
## Description
* There is a need to easily report test tools versions so users can report them to team.
* This PR adds kernel of a functionality to print information about system and tools/Python packages used to run test environment.
* It will help us to diagnose certain issues reported by external and internal tools users.

## Changes:
* Add --info switch, prints system info and exits

## Example

Simple command:
```
$ mbedgt --info
```

or if you want to dump JSON data directly to file ```filename.json```:
```
mbedgt --info > filename.json
```

Prints in JSON formatted output data regarding system and mbed tools packages manes / versions / dependencies:

Example output:
```json
{
    "python": {
        "version": "2.7.11",
        "version_info": "sys.version_info(major=2, minor=7, micro=11, releaselevel='final', serial=0)"
    },
    "mbed-tools": {
        "packages": {
            "mbed-ls": "0.2.6",
            "pyserial": "3.0.1",
            "mbed-host-tests": "0.2.6",
            "mbed-greentea": "0.2.10"
        },
        "details": {
            "mbed-ls": [
                "mbed-ls 0.2.6",
                "prettytable 0.7.2"
            ],
            "pyserial": [
                "pyserial 3.0.1"
            ],
            "mbed-host-tests": [
                "mbed-host-tests 0.2.6",
                "prettytable 0.7.2",
                "pyserial 3.0.1"
            ],
            "mbed-greentea": [
                "mbed-greentea 0.2.10",
                "colorama 0.3.7",
                "mock 1.3.0",
                "lockfile 0.12.2",
                "junit-xml 1.6",
                "mbed-ls 0.2.6",
                "mbed-host-tests 0.2.6",
                "pyserial 3.0.1",
                "prettytable 0.7.2",
                "six 1.10.0",
                "funcsigs 0.4",
                "pbr 1.8.1",
                "six 1.10.0"
            ]
        }
    },
    "os": {
        "uname": [
            "Windows",
            "username",
            "7",
            "6.1.7601",
            "AMD64",
            "Intel64 Family 6 Model 58 Stepping 9, GenuineIntel"
        ],
        "details": {
            "os_win": [
                "7",
                "6.1.7601",
                "SP1",
                "Multiprocessor Free"
            ]
        },
        "os_info": [
            "nt",
            "Windows",
            "7",
            "6.1.7601",
            "win32"
        ]
    }
}
```

### CODE REVIEW

File: mbed_greentea/mbed_info.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+def mbed_get_global_info():
+    info_dict = dict()
+
+    info_dict['os'] = dict()
+    info_dict['mbed-tools'] = dict()
+    info_dict['python'] = dict()
+
+    info_dict['os']['os_info'] = mbed_os_info()
+    info_dict['os']['uname'] = platform.uname()
```
Comment (jupe) [inline mbed_greentea/mbed_info.py]: i think you should take username away from this, its quite private..

File: mbed_greentea/mbed_info.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    info_dict['os']['details'] = dict()
+    info_dict['os']['details']['os_mac'] = platform.mac_ver()
+    info_dict['os']['details']['os_linux'] = platform.linux_distribution()
+    info_dict['os']['details']['os_win'] = platform.win32_ver()
+
+    info_dict['python']['version_info'] = str(sys.version_info)
+    info_dict['python']['version'] =  platform.python_version()
+
+    info_dict['mbed-tools']['packages'] = dict()
```
Comment (jupe) [inline mbed_greentea/mbed_info.py]: why `mbed-tools`, sounds like ARM own all these modules even these are just greentea dependencies

File: mbed_greentea/mbed_info.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+              platform.version(),
+              sys.platform)
+    return result
+
+
+def mbed_get_global_info():
+    info_dict = dict()
+
+    info_dict['os'] = dict()
+    info_dict['mbed-tools'] = dict()
```
Comment (jupe) [inline mbed_greentea/mbed_info.py]: why `mbed-tools`, sounds like ARM own all these modules even these are just greentea dependencies. perhaps this should be just like `greentea-dependensies` or just `dependencies`

**Comment by jupe** [comment]
looks that `os` blocks there is some duplicates, could it be more simple

File: mbed_greentea/mbed_info.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    info_dict['os']['os_info'] = mbed_os_info()
+    info_dict['os']['uname'] = platform.uname()
+
+    info_dict['os']['details'] = dict()
+    info_dict['os']['details']['os_mac'] = platform.mac_ver()
+    info_dict['os']['details']['os_linux'] = platform.linux_distribution()
+    info_dict['os']['details']['os_win'] = platform.win32_ver()
+
+    info_dict['python']['version_info'] = str(sys.version_info)
```
Comment (jupe) [inline mbed_greentea/mbed_info.py]: in example it looks that `python.version.version` doesnt give any more valuable information, so i think it is enough to have just `"python": "2.7.11"`

**Comment by PrzemekWirkus** [comment]
@jupe I've added few modifications, please have a look!
```
$ mbedgt --info
{
    "python": {
        "version": "2.7.11"
    },
    "dependencies": {
        "packages": {
            "mbed-ls": "0.2.6",
            "pyserial": "3.0.1",
            "mbed-host-tests": "0.2.6",
            "mbed-greentea": "0.2.10"
        },
        "details": {
            "mbed-ls": [
                "mbed-ls 0.2.6",
                "prettytable 0.7.2"
            ],
            "pyserial": [
                "pyserial 3.0.1"
            ],
            "mbed-host-tests": [
                "mbed-host-tests 0.2.6",
                "prettytable 0.7.2",
                "pyserial 3.0.1"
            ],
            "mbed-greentea": [
                "mbed-greentea 0.2.10",
                "colorama 0.3.7",
                "mock 1.3.0",
                "lockfile 0.12.2",
                "junit-xml 1.6",
                "mbed-ls 0.2.6",
                "mbed-host-tests 0.2.6",
                "pyserial 3.0.1",
                "prettytable 0.7.2",
                "six 1.10.0",
                "funcsigs 0.4",
                "pbr 1.8.1",
                "six 1.10.0"
            ]
        }
    },
    "os": {
        "details": {
            "os_mac": [
                "",
                [
                    "",
                    "",
                    ""
                ],
                ""
            ],
            "os_win": [
                "7",
                "6.1.7601",
                "SP1",
                "Multiprocessor Free"
            ],
            "os_linux": [
                "",
                "",
                ""
            ]
        },
        "os_info": [
            "nt",
            "Windows",
            "7",
            "6.1.7601",
            "win32"
        ]
    }
}
```

**Comment by jupe** [comment]
Could output be like this:
```
$ mbedgt --info
{
    "application": {
         "name": "mbed-greentea",
         "version": "0.2.10",
         "installation_location": "/Library/Python/2.7/site-packages"
    },
    "python": "2.7.11",
    "dependencies": {
        "packages": {
            "mbed-ls": "0.2.6",
            "pyserial": "3.0.1",
            "mbed-host-tests": "0.2.6",
            "mbed-greentea": "0.2.10"
        },
        "details": {... }
   }
   "platform": {
        'release': '15.4.0', 
        'machine': 'x86_64', 
        'version': 'Darwin Kernel Version 15.4.0: ...', 
        'system': 'Darwin', 
        'processor': 'i386'}
   }
```

I added some more fields like installation_location and processor machine type which might be useful when faced some problem.

platform part could be generated with following way:
```
>>> import platform
>>> plat = { "system":platform.system(), "release": platform.release(),  "version": platform.version(), "machine": platform.machine(), "processor": platform.processor() }
>>> plat
{'release': '15.4.0', 'machine': 'x86_64', 'version': 'Darwin Kernel Version 15.4.0: Fri Feb 26 22:08:05 PST 2016; root:xnu-3248.40.184~3/RELEASE_X86_64', 'system': 'Darwin', 'processor': 'i386'}
```

**Comment by PrzemekWirkus** [comment]
I will close this one for now as we will not implement this feature in near future.

**Pull request closed without merging**